lit_parrot/model.py

```diff
@@ -110,27 +170,41 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         q, k, v = qkv.split(head_size, dim=-1)  # (B, nh, T, hs)
 
         n_elem = int(self.rotary_percentage * head_size)
-        if self.rope_cache is None:
-            self.rope_cache = build_rope_cache(self.block_size, n_elem, x.dtype, x.device)
-        cos, sin = self.rope_cache
-        cos, sin = cos[:T], sin[:T]
 
+        cos, sin = rope
         q_roped = apply_rope(q[..., :n_elem], cos, sin)
         k_roped = apply_rope(k[..., :n_elem], cos, sin)
         q = torch.cat((q_roped, q[..., n_elem:]), dim=-1)
         k = torch.cat((k_roped, k[..., n_elem:]), dim=-1)
 
+        if kv_cache is not None:
+            cache_k, cache_v = kv_cache
+            # check if reached token limit
+            if input_pos[-1] > max_new_tokens:         
```

Comment:
Added token limit logic here cc @carmocca

Reply:
@gkroiz I think there was a mixup between `max_new_tokens` and `max_seq_length`.

- `max_new_tokens` defines how many tokens to generate
- `max_seq_length` defines what is the block size allowed for the model

My comment at https://github.com/Lightning-AI/lit-parrot/pull/51#discussion_r1190554667 referred to the latter.

Also, did you mean "increases overall inference time"?

Reply:
Yes, I meant increases overall inference time, I'll change the original comment

Reply:
fixed the variable discrepancy.